x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/CoordinatorProxyAction.java

```diff
@@ -153,6 +158,66 @@ void handleResponse(List<Slot> slots, MultiSearchResponse response, Exception e)
             }
         }
 
+        static BiConsumer<MultiSearchRequest, BiConsumer<MultiSearchResponse, Exception>> lookupFunction(ElasticsearchClient client) {
+            return (request, consumer) -> {
+                int slot = 0;
+                final Map<String, List<Tuple<Integer, SearchRequest>>> itemsPerIndex = new HashMap<>();
+                for (SearchRequest searchRequest : request.requests()) {
+                    List<Tuple<Integer, SearchRequest>> items =
+                        itemsPerIndex.computeIfAbsent(searchRequest.indices()[0], k -> new ArrayList<>());
+                    items.add(new Tuple<>(slot, searchRequest));
+                    slot++;
+                }
+
+                final AtomicInteger counter = new AtomicInteger(0);
+                final ConcurrentMap<String, Tuple<MultiSearchResponse, Exception>> shardResponses = new ConcurrentHashMap<>();
+                for (Map.Entry<String, List<Tuple<Integer, SearchRequest>>> entry : itemsPerIndex.entrySet()) {
+                    ActionListener<MultiSearchResponse> listener = ActionListener.wrap(
```

Comment:
can you unwrap `entry` to make the code easier to read?

Reply:
I pushed: 511b551033f97b49d2074133bf0e18a11834150e